lib/Reader/readerir.cpp

```diff
@@ -5679,8 +5679,26 @@ IRNode *GenIR::conv(ReaderBaseNS::ConvOpcode Opcode, IRNode *Source) {
     Conversion = SourceIsSigned ? LLVMBuilder->CreateSIToFP(Source, TargetTy)
                                 : LLVMBuilder->CreateUIToFP(Source, TargetTy);
   } else if (SourceTy->isFloatingPointTy() && TargetTy->isIntegerTy()) {
-    Conversion = DestIsSigned ? LLVMBuilder->CreateFPToSI(Source, TargetTy)
-                              : LLVMBuilder->CreateFPToUI(Source, TargetTy);
+    // The ECMA-335 spec says that for unchecked floating point to
+    // integer conversion, that if there is overflow then the result
+    // is unspecified. However for compatibility with other jits
+    // we want to first convert to i64 and then truncate.
+    IntegerType *TargetIntegerType = cast<IntegerType>(TargetTy);
+    int TargetBitWidth = TargetIntegerType->getBitWidth();
+    if (TargetBitWidth < 64) {
+      IntegerType *Int64Ty = Type::getInt64Ty(*JitContext->LLVMContext);
+      if (DestIsSigned) {
+        Conversion = LLVMBuilder->CreateFPToSI(Source, Int64Ty);
+      } else {
+        Conversion = LLVMBuilder->CreateFPToUI(Source, Int64Ty);
+      }
+      // Now do the final conversion to the target type.
```

Comment:
Can you use the ternary operator here for consistency with the 'else' clause below?